### PR TITLE
Update Github workflows

### PR DESCRIPTION
**Pull request by dmitry-ratushnyy**
## Issue
1. Using 5 version of `build_charm_withou_cache` will lead to upload of "flacky" charm
The charm will occasionally fail to check the the installed snaps [with this error](https://chat.canonical.com/canonical/pl/tg7ycf9g37dz8gbpob1pp77ier) leading to a tempory error state

2. Publishing libraries step  was removed accidentally

## Solution

1. Use 5.1.2 version of workflow
2. Restore steps to release library

### CODE REVIEW

**Review by delgod — DISMISSED** [review]

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -67,10 +67,15 @@ jobs:
           github-token: "${{ secrets.GITHUB_TOKEN }}"
 
   build:
-    name: Build charms
-    uses: canonical/data-platform-workflows/.github/workflows/build_charms_with_cache.yaml@v5
+    strategy:
+      fail-fast: true
+      matrix:
+        charms: [".", "tests/integration/relation_tests/new_relations/application-charm"]
```
Comment (carlcsaposs-canonical) [inline .github/workflows/ci.yaml]:
```suggestion
        charm: [".", "tests/integration/relation_tests/new_relations/application-charm"]
```
nit—so it's more readable when using `${{ matrix.charm }}`

File: .github/workflows/release_5_edge.yaml [hunk 2]
```diff
@@ -34,3 +34,18 @@ jobs:
       charmhub-token: ${{ secrets.CHARMHUB_TOKEN }}
     permissions:
       contents: write # Needed to create GitHub release
+
+  release-lubraries:
```
Comment (carlcsaposs-canonical) [inline .github/workflows/release_5_edge.yaml]:
```suggestion
  release-libraries:
```

File: .github/workflows/release_5_edge.yaml [hunk 3]
```diff
@@ -34,3 +34,18 @@ jobs:
...
+    name: Release libraries 
+    needs:
+      - ci-tests
```
Comment (carlcsaposs-canonical) [inline .github/workflows/release_5_edge.yaml:41]:
```suggestion
    needs:
      - ci-tests
      - release
```
consider adding

**Review by carlcsaposs-canonical — DISMISSED** [review]

File: .github/workflows/ci.yaml [hunk 4]
```diff
@@ -67,10 +67,15 @@ jobs:
...
-    uses: canonical/data-platform-workflows/.github/workflows/build_charms_with_cache.yaml@v5
+    strategy:
+      fail-fast: true
+      matrix:
+        charm: [".", "tests/integration/relation_tests/new_relations/application-charm"]
+    name: Build ${{matrix.charms}} charm
+    uses: canonical/data-platform-workflows/.github/workflows/build_charm_without_cache.yaml@v5.1.2
     with:
       charmcraft-snap-channel: "latest/edge"
+      path-to-charm-directory: ${{matrix.charms}}
```
Comment (carlcsaposs-canonical) [inline .github/workflows/ci.yaml]:
```suggestion
      path-to-charm-directory: ${{ matrix.charm }}
```
remember to update this too

**Review by MiaAltieri — DISMISSED** [review]

File: .github/workflows/ci.yaml [hunk 5]
```diff
@@ -67,10 +67,15 @@ jobs:
           github-token: "${{ secrets.GITHUB_TOKEN }}"
 
   build:
-    name: Build charms
-    uses: canonical/data-platform-workflows/.github/workflows/build_charms_with_cache.yaml@v5
+    strategy:
+      fail-fast: true
+      matrix:
+        charm: [".", "tests/integration/relation_tests/new_relations/application-charm"]
+    name: Build ${{matrix.charms}} charm
```
Comment (carlcsaposs-canonical) [inline .github/workflows/ci.yaml]:
```suggestion
    name: Build ${{ matrix.charm }} charm
```

File: .github/workflows/release_5_edge.yaml [hunk 6]
```diff
@@ -34,3 +34,19 @@ jobs:
       charmhub-token: ${{ secrets.CHARMHUB_TOKEN }}
     permissions:
       contents: write # Needed to create GitHub release
+
+  release-libraries:
+    name: Release libraries 
+    needs:
+      - ci-tests
+      - release
```
Comment (carlcsaposs-canonical) [inline .github/workflows/release_5_edge.yaml]:
```suggestion
      - release-charm
```

**Review by delgod — DISMISSED** [review]

**Review by carlcsaposs-canonical — APPROVED** [review]

**Review by MiaAltieri — APPROVED** [review]

**Review by delgod — APPROVED** [review]

**Pull request merged**